packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-endpoint.ts

```diff
@@ -0,0 +1,99 @@
+   */
+  readonly tags?: { [key: string]: string };
+}
+
+/**
+ * A Step Functions Task to create a SageMaker endpoint
+ *
+ * @experimental
+ */
+export class SageMakerCreateEndpoint extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: SageMakerCreateEndpointProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern || sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, SageMakerCreateEndpoint.SUPPORTED_INTEGRATION_PATTERNS);
+    this.taskPolicies = this.makePolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('sagemaker', 'createEndpoint', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject(this.renderParameters()),
+    };
+  }
+
+  private renderParameters(): { [key: string]: any } {
+    return {
+      EndpointConfigName: this.props.endpointConfigName,
+      EndpointName: this.props.endpointName,
+      ...(this.renderTags(this.props.tags)),
+    };
+  }
+
+  private makePolicyStatements(): iam.PolicyStatement[] {
+    const stack = cdk.Stack.of(this);
+    // https://docs.aws.amazon.com/sagemaker/latest/dg/api-permissions-reference.html
+    return [
+      new iam.PolicyStatement({
+        actions: ['sagemaker:createEndpoint'],
+        resources: [
+          stack.formatArn({
+            service: 'sagemaker',
+            resource: 'endpoint',
+            // If the endpoint name comes from input, we cannot target the policy to a particular ARN prefix reliably.
+            // SageMaker uses lowercase for resource name in the arn
+            resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.endpointName) ? '*' : `${this.props.endpointName.toLowerCase()}`,
+          }),
+          stack.formatArn({
+            service: 'sagemaker',
+            resource: 'endpoint-config',
+            // If the endpoint config name comes from input, we cannot target the policy to a particular ARN prefix reliably.
+            // SageMaker uses lowercase for resource name in the arn
+            resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.endpointConfigName) ? '*' : `${this.props.endpointConfigName.toLowerCase()}`,
+          }),
+        ],
+      }),
+      new iam.PolicyStatement({
+        actions: ['sagemaker:ListTags'],
+        resources: ['*'],
+      }),
+    ];
+  }
+
+  private renderTags(tags: { [key: string]: any } | undefined): { [key: string]: any } {
+    return tags ? { Tags: Object.keys(tags).map((key) => ({ Key: key, Value: tags[key] })) } : {};
+  }
```

Comment:
this seems to be duplicated - can we move it into the `.private` folder and reuse it so we can DRY it up a bit?